tests/protocol/test_wire.py

```diff
@@ -1063,6 +1065,26 @@ def test_it_should_return_true_on_success(self):
         with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
             self.assertTrue(protocol.client.try_update_goal_state(), "try_update_goal_state should have succeeded")
 
+    def test_noop_gs_should_skip(self):
+        first_ext = extension_emulator()
```

Comment:
why do we need extensions at all in this test?

Reply:
The intention was to verify that the invalid goal state didn't affect the extensions; now that I think about it, I think there should be a check that no other extension commands were called.

Reply:
we should have a test for the GoalState class here; if you consider this is important enough to add tests specific for extensions we should move this test along with the other extension tests